sqlmesh/dbt/adapter.py

```diff
@@ -290,9 +288,9 @@ def execute(
             self.engine_adapter.fetchdf if fetch else self.engine_adapter.execute  # type: ignore
         )
 
-        expression = parse_one(sql, read=self.engine_adapter.dialect)
+        expression = parse_one(sql, read=self.dialect)
         with _normalize_and_quote(
-            expression, self.engine_adapter.dialect, self.engine_adapter.default_catalog
+            expression, self.dialect, self.engine_adapter.default_catalog  # type: ignore
```

Comment:
Why is the type ignore needed?

Reply:
Dialect is optional in the BaseAdapter class but required by `_normalize_and_quote`. 

We know it will exist here because we infer it from the required engine_adapter passed to the RuntimeAdapter constructor.

For context, we call the RuntimeAdapter and ParsetimeAdaper constructors in [`create_builtin_globals()` in builtin.py](https://github.com/TobikoData/sqlmesh/blob/1acb6efdb22c9e137cd14302332abd212e218f06/sqlmesh/dbt/builtin.py#L338).

Reply:
If you're good with keeping `self.engine_adapter.dialect` here instead of converting to `self.dialect` there's no type issue

Reply:
I slightly prefer the cast, so we are using dialect everywhere, but I'm okay with getting it from engine adapter too.